File: go/remitmd.go
// Package remitmd provides a Go SDK for the remit.md universal AI payment protocol.
//
// remit.md enables AI agents to send and receive USDC payments on EVM L2 chains
// (Base, Arbitrum, Optimism) with support for escrow, streaming, tabs, bounties,
// and deposits.
//
// # Quick Start
//
//	wallet, err := remitmd.NewWallet(os.Getenv("REMITMD_KEY"),
//	    remitmd.WithChain("base"),
//	)
//	if err != nil {
//	    log.Fatal(err)
//	}
//
//	tx, err := wallet.Pay(ctx, "0xRecipient...", decimal.NewFromFloat(1.50),
//	    remitmd.WithMemo("service fee"),
//	)
//
// # Testing
//
// Use MockRemit for unit tests — zero network, zero latency, deterministic:
//
//	mock := remitmd.NewMockRemit()
//	wallet := mock.Wallet()
//
//	tx, err := wallet.Pay(ctx, "0xRecipient", decimal.NewFromFloat(1.50))
//	require.NoError(t, err)
//	assert.True(t, mock.WasPaid("0xRecipient", decimal.NewFromFloat(1.50)))
//
// # Error Handling
//
// All errors are typed *RemitError with a machine-readable Code, actionable Message,
// and link to relevant documentation:
//
//	tx, err := wallet.Pay(ctx, badAddr, amount)
//	var remitErr *RemitError
//	if errors.As(err, &remitErr) {
//	    fmt.Println(remitErr.Code)    // "INVALID_ADDRESS"
//	    fmt.Println(remitErr.DocURL)  // "https://remit.md/docs/errors#INVALID_ADDRESS"
//	}
package remitmd

File: ruby/lib/remitmd.rb
# frozen_string_literal: true

require_relative "remitmd/errors"
require_relative "remitmd/models"
require_relative "remitmd/keccak"
require_relative "remitmd/signer"
require_relative "remitmd/cli_signer"
require_relative "remitmd/http"
require_relative "remitmd/wallet"
require_relative "remitmd/mock"
require_relative "remitmd/a2a"
require_relative "remitmd/x402_client"
require_relative "remitmd/x402_paywall"

# remit.md Ruby SDK - universal payment protocol for AI agents.
#
# @example Direct payment
#   wallet = Remitmd::RemitWallet.new(private_key: ENV["REMITMD_PRIVATE_KEY"])
#   tx = wallet.pay("0xRecipient0000000000000000000000000000001", 1.50)
#   puts tx.tx_hash
#
# @example Using MockRemit in tests
#   mock   = Remitmd::MockRemit.new
#   wallet = mock.wallet
#   wallet.pay("0x0000000000000000000000000000000000000001", 1.00)
#   mock.was_paid?("0x0000000000000000000000000000000000000001", 1.00) # => true
#
module Remitmd
  VERSION = "0.3.0"
end
